server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1197,7 +1197,11 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                     if (state == State.INIT) {
                         // snapshot is still initializing, mark it as aborted
                         shards = snapshotEntry.shards();
-
+                        if (shards.isEmpty()) {
```

Comment:
will this not always be empty in case of INIT state?

Reply:
Removed `if` and added `assert` in https://github.com/elastic/elasticsearch/pull/37686/commits/bc450e851d0ebc70c277a4a167e0b5c1449f57cf :)